File: 99-old-code/7.SpringMvc/in28minutes-services-springmvc/src/main/webapp/resources/js/application/controllers/AddressBookController.js
var AddressBookController = function($scope, $http){

    $scope.editMode = false;
    $scope.position = '';

    $scope.viewAllAddressBook = function(){
        $http.get('address/all.json').success(function(response){
            $scope.addressBooks = response;
        })
    }

    $scope.resetAddressBookField = function(){
        $scope.ab.firstName='';
        $scope.ab.lastName='';
        $scope.ab.phone = '';
        $scope.ab.email = '';
        $scope.editMode = false;
    }

    $scope.addAddressBook = function(ab) {
        $http.post('address/add', ab).success(function(response){
            $scope.viewAllAddressBook();
            $scope.ab.firstName='';
            $scope.ab.lastName='';
            $scope.ab.phone = '';
            $scope.ab.email = '';
        }).error(function(response){
            console.log(response);
        })
    }

    $scope.updateAddressBook = function(ab) {
        $http.put('address/update/'+$scope.position, ab).success(function(response){
            $scope.ab.firstName='';
            $scope.ab.lastName='';
            $scope.ab.phone = '';
            $scope.ab.email = '';
            $scope.viewAllAddressBook();
            $scope.editMode = false;
        }).error(function(response){
            console.log(response);
        })
    }

    $scope.deleteAddressBook = function(id) {
        $http.delete('address/delete/' + id).success(function(response){
            $scope.viewAllAddressBook();
        }).error(function(response){
            console.log(response);
        })
    }

    $scope.deleteAllAddressBook = function(){
        $http.delete('address/delete/all').success(function(response){
            $scope.viewAllAddressBook();
        })
    }

    $scope.editAddressBook = function(pos, addressBook){
        $scope.position = pos;
        $scope.ab = addressBook;
        $scope.editMode = true;
    }

    $scope.viewAllAddressBook();
}

File: 99-old-code/7.SpringMvc/in28minutes-services-springmvc/src/main/java/com/in28minutes/springmvc/services/web/controller/AddressBookController.java
package com.in28minutes.springmvc.services.web.controller;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.ResponseBody;

import com.in28minutes.springmvc.services.web.model.AddressBook;
import com.in28minutes.springmvc.services.web.service.api.AddressBookService;

@Controller
@RequestMapping("/address")
public class AddressBookController {

	@Autowired
	AddressBookService addressBookService;

	@RequestMapping(value = "/all.json", method = RequestMethod.GET)
	public @ResponseBody List<AddressBook> viewAllAddressBook() {
		return addressBookService.viewAllAddressBook();
	}

	@RequestMapping(value = "/add", method = RequestMethod.POST)
	public @ResponseBody void addAddressBookEntry(
			@RequestBody AddressBook addressBook) {
		addressBookService.createAddressBook(addressBook);
	}

	@RequestMapping(value = "/delete/{id}", method = RequestMethod.DELETE)
	public @ResponseBody void deleteAddressBookEntry(
			@PathVariable("id") String id) {
		addressBookService.deleteAddressBook(Integer.valueOf(id));
	}

	@RequestMapping(value = "/update/{pos}", method = RequestMethod.PUT)
	public @ResponseBody void updateAddressBook(
			@RequestBody AddressBook addressBook,
			@PathVariable("pos") String pos) {
		addressBookService.updateAddressBook(Integer.valueOf(pos), addressBook);
	}

	@RequestMapping(value = "/delete/all", method = RequestMethod.DELETE)
	public @ResponseBody void deleteAllAddressBook() {
		addressBookService.deleteAllAddressBook();
	}

	@RequestMapping("/layout")
	public String getTodoPartialPage() {
		return "addressbook/layout";
	}
}
